src/pages/ThorChainLP/queries/hooks/useSendThorTx.tsx

```diff
@@ -0,0 +1,458 @@
+    memo: estimateFeesArgs?.memo ?? '',
+    accountId: estimateFeesArgs?.accountId ?? '',
+    contractAddress: estimateFeesArgs?.contractAddress ?? '',
+    enabled: Boolean(estimateFeesArgs),
+    disableRefetch: Boolean(txId || isSubmitting),
+  })
+
+  const onSignTx = useCallback(async () => {
+    if (
+      !(
+        asset &&
+        wallet &&
+        accountNumber !== undefined &&
+        (dustAmountCryptoBaseUnit || amountCryptoBaseUnit) &&
+        transactionType &&
+        (!isToken(fromAssetId(asset.assetId).assetReference) ||
+          (isToken(fromAssetId(asset.assetId).assetReference) && inboundAddressData))
+      )
+    )
+      return
+
+    const { account } = fromAccountId(accountId)
+
+    await (async () => {
+      switch (transactionType) {
+        case 'MsgDeposit': {
+          if (!estimateFeesArgs) throw new Error('No estimateFeesArgs found')
+          if (accountNumber === undefined) throw new Error(`No account number found`)
+          if (!memo) return undefined
+
+          const adapter = assertGetThorchainChainAdapter()
+          const estimatedFees = await estimateFees(estimateFeesArgs)
+
+          // LP deposit using THOR is a MsgDeposit tx
+          const { txToSign } = await adapter.buildDepositTransaction({
+            from: account,
+            accountNumber,
+            value: shouldUseDustAmount ? dustAmountCryptoBaseUnit! : amountCryptoBaseUnit!,
+            memo,
+            chainSpecific: {
+              gas: (estimatedFees as FeeDataEstimate<KnownChainIds.ThorchainMainnet>).fast
+                .chainSpecific.gasLimit,
+              fee: (estimatedFees as FeeDataEstimate<KnownChainIds.ThorchainMainnet>).fast.txFee,
+            },
+          })
+          const signedTx = await adapter.signTransaction({
+            txToSign,
+            wallet,
+          })
+          const _txId = await adapter.broadcastTransaction({
+            senderAddress: account,
+            receiverAddress: THORCHAIN_POOL_MODULE_ADDRESS,
+            hex: signedTx,
+          })
+          const _txIdLink = getTxLink({
+            defaultExplorerBaseUrl: 'https://viewblock.io/thorchain/tx/',
+            txId: _txId ?? '',
+            name: SwapperName.Thorchain,
+          })
+
+          toast({
+            title: translate('modals.send.transactionSent'),
+            description: _txId ? (
+              <Text>
+                <Link href={_txIdLink} isExternal>
+                  {translate('modals.status.viewExplorer')} <ExternalLinkIcon mx='2px' />
+                </Link>
+              </Text>
+            ) : undefined,
+            status: 'success',
+            duration: 9000,
+            isClosable: true,
+            position: 'top-right',
+          })
+
+          setTxId(_txId)
+          setSerializedTxIndex(
+            serializeTxIndex(accountId, _txId, account, { memo, parser: 'thorchain' }),
+          )
+
+          break
+        }
+        case 'EvmCustomTx': {
```

Comment:
using evm as an example this if we continue unifying things, I would expect this function to look like just calling the appropriate `buildAndBroadcast` function (assuming the buildTxInput is already available and constructed so we don't have to redo),  and then fall through and every chain would just do the same txlink + toast, etc

Reply:
See https://github.com/shapeshift/web/pull/6700#discussion_r1569323927 for next planned steps (not in this PR obviously)